### PR TITLE
Adding D11 support for nextjs.

### PR DESCRIPTION
**Pull request by apathak18**
This pull request is for: (mark with an "x")

- [ ] `examples/*`
- [x] `modules/next`
- [ ] `packages/next-drupal`
- [ ] `starters/basic-starter`
- [ ] `starters/graphql-starter`
- [ ] `starters/pages-starter`
- [ ] Other

GitHub Issue: #
_Please add a link to the GitHub issue: https://github.com/chapter-three/next-drupal/issues/777
where this problem is discussed._
Getting ready for this 
- [ ] I need help adding tests. (mark with an "x")
      _Code changes need test coverage. If you don't know
      how to make tests, check this box to ask for help._

## Describe your changes

A clear and concise description of what the request is.

If applicable, add screenshots to help explain your issue.

### CODE REVIEW

**Comment by apathak18** [comment]
@shadcn I'm not able to add the reviewer and other pipelines are blocked without reviewer.
Could you please review the PR and check if any changes required to support D11.

File: modules/next/composer.json [hunk 1]
```diff
@@ -12,11 +12,14 @@
   ],
   "require": {
     "drupal/decoupled_router": "^2.0",
+    "drupal/graphql": "^4.8",
+    "drupal/graphql_compose": "^2.2",
+    "drupal/jwt": "^2 || ^3",
```
Comment (JohnAlbin) [inline modules/next/composer.json:17]: This seems unrelated to the D11 upgrade.

**Review by JohnAlbin — CHANGES_REQUESTED** [review]
After a quick review, this is the only thing I see as an issue.

**Comment by apathak18** [comment]
@JohnAlbin,
Validated with these changes and it is running the site without any error!!
Content syncing and image rendering ~~ attaching the screenshots for the same
Screenshot 2024-08-03 at 6 45 01 PM
Screenshot 2024-08-03 at 6 44 52 PM

File: modules/next/composer.json [hunk 2]
```diff
@@ -17,6 +17,6 @@
     "drupal/subrequests": "^3.0"
   },
   "require-dev": {
-    "phpunit/phpunit": "^6.0 || ^7.0 || ^8.0 || ^9.0"
+    "phpunit/phpunit": "^8.0 || ^9.0 || ^10"
```
Comment (vishalkhode1) [inline modules/next/composer.json:20]: @apathak18 I don't think we need phpunit 8, as I believe Drupal 10 works on PHPunit 9.5 and above.

File: modules/next/modules/next_graphql/next_graphql.info.yml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 name: 'Next.js GraphQL'
 type: module
 description: 'GraphQL for Next.js'
-core_version_requirement: ^9 || ^10
+core_version_requirement: ^10.1 || ^11
```
Comment (vishalkhode1) [inline modules/next/modules/next_graphql/next_graphql.info.yml:4]: @apathak18 Can we keep `core_version_requirement` uniform across all module/sub-modules. Ex: Here we've added `^10.1` and in `next_extras` sub-module we have' added `^10`.

File: modules/next/tests/modules/next_tests/next_tests.info.yml [hunk 4]
```diff
@@ -1,7 +1,7 @@
 name: Next.js tests
 description: Configures testing for Next.js
 type: module
+core_version_requirement: ^10 || ^11
```
Comment (vishalkhode1) [inline modules/next/tests/modules/next_tests/next_tests.info.yml:4]: @apathak18 I think we can remove `core_version_requirement` key all-together. As this is not required for tests module. Re: https://www.drupal.org/node/3096609

**Review by vishalkhode1 — CHANGES_REQUESTED** [review]

File: modules/next/tests/modules/next_tests/next_tests.info.yml (hunk 4 above)
Comment (apathak18) [reply modules/next/tests/modules/next_tests/next_tests.info.yml]: I guess we need to keep core_version_requirement  here https://github.com/apathak18/next-drupal/actions/runs/10260927615/job/28387772151#step:12:182

**Comment by apathak18** [comment]
I've refractor the tests to move delete action log after revalidate as I've noticed it is logging after revalidating (Reason delete event fire on hook_entity_predelete)
Screenshot 2024-08-06 at 2 12 00 PM

Tests output for D11
Screenshot 2024-08-06 at 2 29 08 PM

**Comment by apathak18** [comment]
@JohnAlbin Please review this PR and let me know if any other changes required.

File: .github/workflows/next.yml [hunk 5]
```diff
@@ -22,15 +22,21 @@ jobs:
...
+          - "11.0.x"
         exclude:
           - drupal: "10.0.x"
             php: "8.3"
           - drupal: "10.1.x"
             php: "8.3"
+          - drupal: "11.0.x"
+            php: "8.1"
+          - drupal: "11.0.x"
+            php: "8.2"
```
Comment (yobottehg) [inline .github/workflows/next.yml:35]: 11.0 required PHP 8.3

Comment (apathak18) [reply .github/workflows/next.yml:35]: @yobottehg Correct!!
Pipeline is running on 8.3
If configuration is to exclude running with 8.1, 8.2 versions of php for Drupal 11

File: modules/next/tests/src/Kernel/Event/EntityActionEventTest.php [hunk 6]
```diff
@@ -63,39 +62,36 @@ public function testEntityActionEvents() {
...
-      ->fields('wd', ['message', 'variables'])
+  protected function assertLogMessage(string $action) {
+    $message = "Event @event dispatched for entity @label and action @action.";
+    $variables = [
+      '@event' => 'next.entity.action',
+      '@label' => 'A page',
+      '@action' => $action,
+    ];
+
+    $this->assertNotEmpty(Database::getConnection()->select('watchdog', 'w')
```
Comment (vishalkhode1) [inline modules/next/tests/src/Kernel/Event/EntityActionEventTest.php:87]: @apathak18 I'm not sure on this, but we can use it like below instead of using `Database` class:
```suggestion
    $this->assertNotEmpty($this->container->get('database')->select('watchdog', 'w')
```

**Review by vishalkhode1 — APPROVED** [review]

**Review by rajeshreeputra — APPROVED** [review]

**Review by backlineint — APPROVED** [review]
Generally things look good here. Thanks to all who contributed on this one.

In general, I'd encourage the scope of an upgrade issue like this to be kept just to things necessary for the upgrade - core version requirement in module info files, any deprecated code. There were some things along the way outside of that, and I still think some of the changes to the NextSettingsForm are not related to D11 compatibility. Don't doubt that these could be worthwhile changes, but keeping them in a separate PR will hopefully make it easier for us to get important things Drupal version compatibility through more quickly.

I confirmed that the module tests are passing locally, and also did some manual e2e testing.

There are still sone deprecation warnings thrown by the module tests, but I'll create a follow up PR for that so we can merge this.

File: modules/next/composer.json (hunk 1 above)
Comment (backlineint) [reply modules/next/composer.json]: This has been resolved.

**Comment by backlineint** [comment]
This has been released as 2.0.0-beta1: https://www.drupal.org/project/next/releases/2.0.0-beta1
